src/main/java/nextstep/fp/Lambda.java

```diff
@@ -27,30 +27,23 @@ public void run() {
     }
 
     public static int sumAll(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            total += number;
-        }
-        return total;
+        return sumByCondition(numbers, number -> true );
     }
 
     public static int sumAllEven(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number % 2 == 0) {
-                total += number;
-            }
-        }
-        return total;
+        return sumByCondition(numbers, number -> number % 2 == 0);
     }
 
     public static int sumAllOverThree(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number > 3) {
-                total += number;
-            }
-        }
-        return total;
+        return sumByCondition(numbers, number -> number > 3);
+    }
+
+    public static int sumByCondition(List<Integer> numbers, Conditional condition){
+        return numbers
+                .stream()
+                .filter(num -> condition.test(num))
+                .reduce(Integer::sum)
+                .get()
```

Comment:
```suggestion
                .reduce(0, Integer::sum)
```
이렇게 하면 Optional로 반환되지 않습니다!